src/lib/cert/x509/x509_ext.cpp

```diff
@@ -136,7 +146,7 @@ void Extensions::decode_from(BER_Decoder& from_source)
 
       m_extensions_raw.emplace(oid, std::make_pair(value, critical));
 
-      std::unique_ptr<Certificate_Extension> ext(get_extension(oid));
+      std::unique_ptr<Certificate_Extension> ext(get_extension(oid, critical));
 
       if(!ext && critical && m_throw_on_unknown_critical)
```

Comment:
Because of `get_extension()` now returning an `Unknown_Critical_Extension`, this check does not work anymore. I wonder if this functionality is neither desired nor needed anymore?